src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java

```diff
@@ -189,13 +193,13 @@ public CalledHaplotypes callMutations(
 
     // compute the likelihoods of: AA, AB, AC. . . where A is ref and B, C. . . are somatic alts
     private PerAlleleCollection<Double> getHetGenotypeLogOdds(ReadLikelihoods<Allele> likelihoods,
-                                                              final boolean isNormal,
-                                                              final Strand strand) {
-        final String sample = isNormal ? matchedNormalSampleName : tumorSampleName;
-        final OptionalDouble givenAltAlleleFraction = isNormal ? GERMLINE_HET_ALT_FRACTION : NO_FIXED_TUMOR_ALT_FRACTION;
-        final PerAlleleCollection<Double> hetGenotypeLogLks = getHetGenotypeLogLikelihoods(likelihoods, sample, givenAltAlleleFraction, strand);
+                                                              final boolean isGermline,
+                                                              final Strand strand, final String sample) {
+        final OptionalDouble givenAltAlleleFraction = isGermline ? GERMLINE_HET_ALT_FRACTION : NO_FIXED_TUMOR_ALT_FRACTION;
+        final String sampleForAlleleFractions = (sample == matchedNormalSampleName && isGermline) ? matchedNormalSampleName : tumorSampleName;
```

Comment:
I did case 3) this way intentionally, the idea being that the artifact might not have the same allele fraction in the normal as in the tumor, but its presence in the normal is still useful evidence.  The way it works now is "If there were no tumor sample and we pretended the normal was a tumor, would we call this allele?"